hist/hist/src/TH1.cxx

```diff
@@ -741,7 +741,7 @@ TH1::TH1(const char *name,const char *title,Int_t nbins,const Double_t *xbins)
 
 Bool_t TH1::AddDirectoryStatus()
 {
-   return fgAddDirectory;
+   return !ROOT::ROOT7MemoryManagement() && fgAddDirectory;
```

Comment:
Note that this flag is primarily about file constructions (and secondarily about shared ownership).  Beside the (intentional) memory leak it introduces in existing scripts, it leads most of them to not longer write the histograms to the file (because they are no longer attached to the File).  It is not clear whether this kind of approach is worth it compared to a more user active change (ie. Switching explicitly from TFile to RFile).